### PR TITLE
New detector: Library should not have public or external functions

### PR DESCRIPTION
**Pull request by TilakMaddy**
It's not ideal to have public or external functions in libraries as they are not designed to interact with other external contracts and or maintain state. Hence it can be misleading.

### CODE REVIEW

File: tests/contract-playground/src/LibraryContract.sol [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                x = (y / x + x) / 2;
+            }
+        } else if (y != 0) {
+            z = 1;
+        }
+        // else z = 0 (default value)
+    }
+    // BAD
+    function remove(uint256[] storage arr, uint256 index) public {
+        // Move the last element into the place to delete
```
Comment (TilakMaddy) [inline tests/contract-playground/src/LibraryContract.sol:21]: Example here

File: aderyn_core/src/detect/nc/library_has_public_external_funcs.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            .filter(|x| x.kind == ContractKind::Library)
+        {
+            for function in contract.function_definitions() {
+                if function.visibility == Visibility::Public
+                    || function.visibility == Visibility::External
+                {
+                    capture!(self, context, function);
+                }
+            }
+        }
```
Comment (TilakMaddy) [inline aderyn_core/src/detect/nc/library_has_public_external_funcs.rs:31]: Logic here

**Comment by alexroan** [comment]
I'm not convinced this is worthy of being something that Aderyn recommends as a security concern right now. Let's shelf this.

**Comment by TilakMaddy** [comment]
Got it : ) Turns out I have a reference for this one - 

(LightChaser's findings for GOAT.TECH Cantina competition - https://cantina.xyz/competitions/f214cf86-cc80-40c0-a70b-e9bb25d7ac80)

But can I conclude that  Aderyn's flavor differs slightly from what we see in bot reports for competitions?

**Pull request closed without merging**